source/aggregation/agg-exp-ops.txt

```diff
@@ -0,0 +1,1185 @@
+       | `getMap() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#getMap(java.lang.String)>`__
+       | `getNumber() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#getNumber(java.lang.String)>`__
+       | `getString() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#getString(java.lang.String)>`__
+     - :manual:`$getField </reference/operator/aggregation/getField/>`
+
+   * - `hasField() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#hasField(java.lang.String)>`__
+     - *No corresponding operator*
+
+   * - `merge() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#merge(com.mongodb.client.model.mql.MqlDocument)>`__
+     - :manual:`$mergeObjects </reference/operator/aggregation/mergeObjects/>`
+
+   * - `setField() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#setField(java.lang.String,com.mongodb.client.model.mql.MqlValue)>`__
+     - :manual:`$setField </reference/operator/aggregation/setField/>`
+
+   * - `unsetField() <{+core-api+}/com/mongodb/client/model/mql/MqlDocument.html#unsetField(java.lang.String)>`__
+     - :manual:`$unsetField </reference/operator/aggregation/unsetField/>`
+
+Suppose you have a collection of legacy customer data which includes
+addresses as child documents under the ``mailing.address`` field. You want
+to find all the customers who live in Washington state. A
+document in this collection might resemble the following:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "_id": ...,
+     "customer.name": "Mary Kenneth Keller",
+     "mailing.address":
+     {
+       "street": "601 Mongo Drive",
+       "city": "Vasqueztown",
+       "state": "CO",
+       "zip": 27017
+     }
+   }
+
+The ``getDocument()`` method retrieves the ``mailing.address`` field as a
+document so the nested ``state`` field can be retrieved with the
+``getString()`` method. The ``eq()`` method checks if the value of the
+``state`` field is ``"WA"``.
+
+The following code shows the pipeline for this aggregation:
+
+.. literalinclude:: /includes/aggregation/aggExpressions.kt
+   :start-after: start-document-aggregation
+   :end-before: end-document-aggregation
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+       { $match: {
+           $expr: {
+               $eq: [{
+                   $getField: {
+                       input: { $getField: { input: "$$CURRENT", field: "mailing.address"}},
+                       field: "state" }},
+                   "WA" ]
+   }}}]
+
+Map Operations
+~~~~~~~~~~~~~~
+
+You can perform a map operation on a value of either type ``MqlMap`` or
+``MqlEntry`` using the methods described in this section.
+
+.. tip::
+
+   Represent data as a map if the data maps
+   arbitrary keys such as dates or item IDs to values.
```

Comment:
Thinking about the purpose of this page, is this tip necessary? This seems more relevant for a page about how to structure a data schema, as opposed to performing aggregation operations. We also have a pretty thorough example usecase later on in the section.

Reply:
can remove